modules/save_data.rb

```diff
@@ -0,0 +1,60 @@
+require './student'
+require './teacher'
+require './rental'
+module SaveData
+  def save_data(books, persons, rentals)
+    save_books(books)
+    save_persons(persons)
+    save_rentals(rentals)
+  end
+
+  def save_books(books)
+    saved_books = []
+    books.each do |book|
+      saved_books << {
+        title: book.title,
+        author: book.author
+      }
+    end
+    ruby = JSON.generate(saved_books)
+    File.write('./data/books.json', ruby)
+  end
+
+  def save_persons(persons)
+    saved_persons = []
+    persons.each_with_index do |person, _index|
+      saved_persons << if person.instance_of? Student
+                         {
```

Comment:
- [ ] Great work on implementing the change of saving data in files. But previous reviewer asked you to handle the scenario in which you won't have json files already created. :thinking:

**Explanation :notebook:**
- First delete the json date files
- Then run your ruby application by ruby main.rb
You won't be able to start your application because you don't have json files.

**Fix**
Make a function which will check if the json files exist or not when starting the application. If exist then its ok otherwise it will create json files with empty array in it. :+1: